src/turbomind/kernels/gemm_s_f16/gemm_s4_f16.cu

```diff
@@ -41,6 +41,16 @@ struct SiluActivation {
     }
 };
 
+struct Add {
+    static __inline__ __device__ void apply(uint data, int m, int n, half* C, int M, int N)
+    {
+        if (n < N) {
+            C[n * M + m] += ((half2&)data).x;
+            C[n * M + m + 1] += ((half2&)data).y;
+        }
```

Comment:
this is weird since the following implementation failed:
```cpp
        if (n < N) {
            (half2&)C[n * M + m] += (half2&)data;
        }
```

Reply:
@lzhangzz

Reply:
In my test 
```
if (n < N) { 
    (half2&)C[n * M + m] += (half2&)data; 
}
```
works fine.